engine/include/logger/log.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <logger/target.hpp>
+#include <source_location>
+#include <string_view>
+
+namespace gen::logger {
+using SrcLoc = std::source_location;
+
+void error(std::string_view category, std::string_view message, SrcLoc const& location = SrcLoc::current());
+void warn(std::string_view category, std::string_view message, SrcLoc const& location = SrcLoc::current());
+void info(std::string_view category, std::string_view message, SrcLoc const& location = SrcLoc::current());
+void debug(std::string_view category, std::string_view message, SrcLoc const& location = SrcLoc::current());
```

Comment:
I'd like your thoughts on this, but even though allowing the user to explicitly type the category, I fear it could be bug-prone if a user fails to enter the consistent category. For example, what if one user puts [Rendering] and another puts [Graphics]?

This could possibly create issues with category filtering down the line. As trusting the developer to always do the right thing is generally not the best choice. :P

Personally, I think we should instead define log categories as an enum or something explicit that can be added too when we need more categories versus letting the user define it explicitly.

Reply:
Adding `static constexpr std::string_view category_v{"Graphics"}` and enforcing logs in that domain to use it as the category argument solves that problem. Using enums does make it _impossible_ to pass in an invalid category, but at the cost of two things I think are worth keeping:
- Having to do zero maintenance on any other files when adding a new category to the codebase.
- Not having to prefix `logger::Category::eGraphics` (but instead `graphics_v`).

Reply:
The string_view solution does seem like a valid option. @Kouros26 , what is your opinion?

Reply:
I also prefer the string_view option now that I've read karnage's explanation